exchanges/coinbasepro/coinbasepro_wrapper.go

```diff
@@ -792,18 +792,28 @@ func (c *CoinbasePro) GetOrderHistory(ctx context.Context, req *order.GetOrdersR
 		return nil, err
 	}
 	var respOrders []GeneralizedOrderResponse
-	for i := range req.Pairs {
-		fpair, err := c.FormatExchangeCurrency(req.Pairs[i], asset.Spot)
-		if err != nil {
-			return nil, err
+	if len(req.Pairs) > 0 {
+		for i := range req.Pairs {
+			fpair, err := c.FormatExchangeCurrency(req.Pairs[i], asset.Spot)
+			if err != nil {
+				return nil, err
+			}
+			resp, err := c.GetOrders(ctx,
+				[]string{"done"},
+				fpair.String())
+			if err != nil {
+				return nil, err
+			}
+			respOrders = append(respOrders, resp...)
 		}
+	} else {
 		resp, err := c.GetOrders(ctx,
```

Comment:
Seems like this could just become 
```suggestion
		respOrders, err := c.GetOrders(ctx,
```
And remove the line `respOrders = resp` below

Reply:
Hi, unfortunately just `respOrders, err := c.GetOrders(ctx,` would not work as respOrders := would mean declaring a new variable, go would complain "unused variable 'respOrders'" if not assigning it back to the original variable respOrders, to make c.GetOrders to be directly assigned to the original respOrders, would need to declare err first:

```
var err error
respOrders, err = c.GetOrders(ctx,
```

But to align with the existing style, I went with `resp, err := c.GetOrders(ctx,` then assign it back to respOrders

Reply:
Thanks for the explanation! Yeah I missed the `:=` part of my suggest regarding the `err` declaration. I'm happy with how you've done it 🎉